src/Functions/generateUUIDv4.cpp

```diff
@@ -93,7 +93,48 @@ class FunctionGenerateUUIDv4 : public TargetSpecific::Default::FunctionGenerateU
 
 REGISTER_FUNCTION(GenerateUUIDv4)
 {
-    factory.registerFunction<FunctionGenerateUUIDv4>();
+    /// generateUUIDv4 documentation
+    FunctionDocumentation::Description description_generateUUIDv4 = R"(Generates a [version 4](https://tools.ietf.org/html/rfc4122#section-4.4) [UUID](../data-types/uuid.md).)";
+    FunctionDocumentation::Syntax syntax_generateUUIDv4 = "generateUUIDv4([expr])";
+    FunctionDocumentation::Arguments arguments_generateUUIDv4 = {
+        {"expr", "An arbitrary expression used to bypass common subexpression elimination if the function is called multiple times in a query. The value of the expression has no effect on the returned UUID. Optional."}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_generateUUIDv4 = {"Returns a UUIDv4.", {"UUID"}};
+    FunctionDocumentation::Examples examples_generateUUIDv4 = {
+    {
+        "Usage example",
+        R"(
+CREATE TABLE tab (uuid UUID)
+ENGINE = MergeTree
+ORDER BY tuple();
+
+INSERT INTO tab SELECT generateUUIDv4();
+
+SELECT * FROM tab;
+        )",
+        R"(
+┌─────────────────────────────────uuid─┐
+│ f4bf890f-f9dc-4332-ad5c-0c18e73f28e9 │
+└──────────────────────────────────────┘
+        )"
+    },
+    {
+        "multiple UUIDs generated per row",
```

Comment:
I think this example (which demonstrates the optional argument) would also be nice to have for generateUUIDv7.cpp.